src/crosisdownload/mod.rs

```diff
@@ -58,37 +58,82 @@ static NO_GO: [&str; 28] = [
 
 const MAX_FILE_PARALLELISM: usize = 20;
 
+#[derive(Clone)]
 pub struct DownloadLocations {
     pub main: String,
     pub git: String,
     pub staging_git: String,
     pub ot: String,
 }
 
-enum FilePath {
-    Cont(String),
-    Break,
+pub enum DownloadStatus {
+    Full,
+    NoHistory,
+}
+
+#[derive(Clone, Copy)]
+pub struct ReplInfo<'a> {
+    pub id: &'a str,
+    pub username: &'a str,
+    pub slug: &'a str,
 }
 
 pub async fn download(
     client: reqwest::Client,
-    replid: String,
-    replname: &str,
+    replinfo: ReplInfo<'_>,
+    download_zip: &str,
```

Comment:
```suggestion
    download_zip_filename: &str,
```

(I wasn't immediately sure what it was by looking at the function signature)

Reply:
LGTM, tho make sure that all uses of it get renamed

Reply:
eh i'll just add a doc comment